docs/Training_data_sources.md

```diff
@@ -0,0 +1,98 @@
+
+Of the 26 multiple sources only 3 were not in agreement:
+- 3 times the grant description was labelled 'not tech', but RF data labelled as 'tech'
+- 0 times the grant description was labelled 'not tech', but EPMC data labelled as 'tech'
+- 3 times the grant description and RF labels both said tech
+- 13 times the grant description and EPMC labels both said tech
+- 7 times both EPMC and RF labels said tech
+
+# 2. How well does the 210223 ensemble do for each of these 4 sources?
+
+Using the test data (i.e. not any data that went into training the models), the 210223 ensemble model performs differently for data points labelled from the different sources.
+
+- Of the 19 ResearchFish labelled data points (all labelled as tech) **0.632** were correctly labelled as tech when using the grant description.
+- Of the 35 EPMC labelled data points (all labelled as tech) **0.714** were correctly labelled as tech when using the grant description.
+
+When tagging a grant as tech or not from the original grant descriptions the model performs better:
+- 114 original grants via Excel: 'precision': 0.805, 'recall': **0.892**, 'f1': 0.846
+- 73 grants tagged via Prodigy: 'precision': 0.771, 'recall': **0.902**, 'f1': 0.831
+- 187 grants **either** tagged via Excel or Prodigy: 'precision': 0.787, 'recall': **0.897**, 'f1': 0.838
+
+Ensemble performance on all test data (recap):
+- 241 grants: 'precision': 0.849, 'recall': 0.811, 'f1': 0.829
+
+# 3. Does training the models with no EPMC or RF training data contributions improve the model?
+
+In the notebook I output `../data/prodigy/merged_tech_grants/merged_tech_grants_noepmcrf.jsonl` which is the Prodigy outputted data (`'../data/prodigy/merged_tech_grants/merged_tech_grants.jsonl'`) minus the additions from the EPMC or RF tagged training data (from the `training_data/210126/training_data.csv` training data).
+
+This can be run in `prodigy_training_data.py` to create a new training data csv:
+```
+python -i nutrition_labels/prodigy_training_data.py --prodigy_data_dir 'data/prodigy/merged_tech_grants/merged_tech_grants_noepmcrf.jsonl'
+```
+which outputted `training_data/210305/training_data.csv`. 
+
+| Tag code | Meaning | Number of grants - 210221 | Number of grants - 210305 |
+|---|---|--- |--- |
+| 1 | Relevant | 495 | 306 |
+| 0 | Not relevant | 485 | 484 |
+
+Then I start running models again:
+```
+python -i nutrition_labels/grant_tagger.py --training_data_file 'data/processed/training_data/210305/training_data.csv' --vectorizer_type tfidf --model_type naive_bayes
+```
+The split seed was set to 7 - but this time I didn't run any of the 'best seed' experiments - I just kept it set to what it was at before.
+
+| Date | Vectorizer type | Model type | Bert type (if relevant) | Train F1 | Test F1 | Test precision | Test recall |
+| --- | --- | --- | --- | --- | --- | --- | --- |
+| 210221 | count | log_reg | - | 1.000 | 0.772 | 0.806 | 0.741 |
+| 210305 | count | log_reg | - | 1.000 | 0.805 | 0.831 | 0.780 |
```

Comment:
The date got me confused because I usually use YYYY-MM-DD 🙈

Reply:
Same here